MonoGame.Framework/Random.cs

```diff
@@ -0,0 +1,386 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Xna.Framework
+{
+    public static partial class MathHelper
+    {
+        /// <summary>
+        /// Class for generating random numbers consistently when using Mono or Microsoft compilers.
+        /// </summary>
+        public class Random
+        {
+            #region Static methods
+
+            /// <summary>
+            /// Utility method for generating a seed for the random number generator.
+            /// </summary>
+            [CLSCompliant(false)]
+            public static ulong GenerateSeed()
+            {
+                byte[] guid = Guid.NewGuid().ToByteArray();
+                return (BitConverter.ToUInt64(guid, 0) ^ BitConverter.ToUInt64(guid, 7));
+            }
+
+            #endregion
+
+            #region Private Fields
+
+            private ulong _seed;
+            private ulong _state;
+            private ulong _seq;
+
+            private const float UINT32_TO_FLOAT = 1.0f / ((uint.MaxValue >> 8) + 1);
+            private const double UINT64_TO_DOUBLE = 1.0 / ((ulong.MaxValue >> 11) + 1);
+
+            #endregion
+
+            #region Properties
+
+            /// <summary>
+            /// Original seed for the random number generator. Assigning to this will 
+            /// reset the RNG to this state as well as assigning the seed.
+            /// </summary>
+            [CLSCompliant(false)]
```

Comment:
We get compiler warnings if a public API uses unsigned types unless the `CLSCompliant` attribute is applied to it.